recipes/google-cloud-cpp-core/meta.yaml

```
@@ -0,0 +1,360 @@
+{% set version = "2.17.0" %}
+
+{%- macro test_subpackage(feature) %}
+commands:
+  - test -f $PREFIX/lib/libgoogle_cloud_cpp_{{ feature }}.{{ version }}.dylib  # [osx]
+  - test -f $PREFIX/lib/libgoogle_cloud_cpp_{{ feature }}.so.{{ version }}  # [linux]
+  - test ! -f $PREFIX/lib/pkgconfig/google_cloud_cpp_{{ feature }}.pc  # [not win]
+  - test ! -f $PREFIX/include/google/cloud/version.h  # [not win]
+  - test ! -f $PREFIX/lib/cmake/google_cloud_cpp_{{ feature }}/google_cloud_cpp_{{ feature }}-config.cmake  # [not win]
+  - if exist %LIBRARY_PREFIX%\\include\\google\\cloud\\version.h exit 1  # [win]
+  - if exist %LIBRARY_PREFIX%\\lib\\cmake\\google_cloud_cpp_{{ feature }}\\google_cloud_cpp_{{ feature }}-config.cmake exit 1  # [win]
```

Suggest to restructure this a bit; you also don't need the double backslash on windows
```suggestion
  # presence of shared library (unix)
  - test -f $PREFIX/lib/libgoogle_cloud_cpp_{{ feature }}.{{ version }}.dylib   # [osx]
  - test -f $PREFIX/lib/libgoogle_cloud_cpp_{{ feature }}.so.{{ version }}      # [linux]
  # presence of static library (windows)
  - if not exist %LIBRARY_LIB%\libgoogle_cloud_cpp_{{ feature }}.lib exit 1     # [win]

  # presence of headers
  - test ! -f $PREFIX/include/google/cloud/version.h                            # [unix]
  - if exist %LIBRARY_INC%\google\cloud\version.h exit 1                        # [win]

  # absence of metadata for CMake & pkgconfig (belongs in devel package)
  - test ! -f $PREFIX/lib/pkgconfig/google_cloud_cpp_{{ feature }}.pc           # [unix]
  - test ! -f $PREFIX/lib/cmake/google_cloud_cpp_{{ feature }}/google_cloud_cpp_{{ feature }}-config.cmake          # [unix]
  - if exist %LIBRARY_LIB%\cmake\google_cloud_cpp_{{ feature }}\google_cloud_cpp_{{ feature }}-config.cmake exit 1  # [win]
```

---

Sidenote: the header checks don't relate to `feature`, so they should be done separately from the macro in the testing of `libgoogle-cloud`.

---

Done.  Maybe we should remove these header tests, as there is no easy way to get a per-feature header.